src/main/java/com/microsoft/sqlserver/jdbc/SQLServerDataTable.java

```diff
@@ -152,34 +152,31 @@ private void internalAddrow(JDBCType jdbcType,
             Object val,
             Object[] rowValues,
             Map.Entry<Integer, SQLServerDataColumn> pair) throws SQLServerException {
-
-        SQLServerDataColumn currentColumnMetadata = pair.getValue();
-        boolean isColumnMetadataUpdated = false;
-        boolean bValueNull;
-        int nValueLen;
-        switch (jdbcType) {
-            case BIGINT:
-                rowValues[pair.getKey()] = (null == val) ? null : Long.parseLong(val.toString());
-                break;
-
-            case BIT:
-                rowValues[pair.getKey()] = (null == val) ? null : Boolean.parseBoolean(val.toString());
-                break;
-
-            case INTEGER:
-                rowValues[pair.getKey()] = (null == val) ? null : Integer.parseInt(val.toString());
-                break;
-
-            case SMALLINT:
-            case TINYINT:
-                rowValues[pair.getKey()] = (null == val) ? null : Short.parseShort(val.toString());
-                break;
-
-            case DECIMAL:
-            case NUMERIC:
-                BigDecimal bd = null;
-                if (null != val) {
-                    bd = new BigDecimal(val.toString());
+    	if(null != val) {
+	        SQLServerDataColumn currentColumnMetadata = pair.getValue();
+	        boolean isColumnMetadataUpdated = false;
+	        int nValueLen;
+	        switch (jdbcType) {
+	            case BIGINT:
+	                rowValues[pair.getKey()] = (val instanceof Number ? (Long)val : Long.parseLong(val.toString()));
```

Comment:
Should this really check for `Number` and then cast to `Long`? I would either change to `instanceof Long` or use `((Number)val).longValue()` (which may introduce an unnecessary reboxing if `val` is `Long`)